refactor the documentation, testing and building to be consistent

ikethecoder opened the pull request.

ikethecoder committed:
start on some docs

ikethecoder committed:
start on some docs

ikethecoder committed:
Merge branch 'cleanup' of https://github.com/bcgov/kong-oss-plugins into cleanup

ikethecoder committed:
add testsuite with playwright for automated testing

ikethecoder committed:
tests for oidc plugin

ikethecoder committed:
cleanup unused bits and expand on tests

ikethecoder committed:
fix github action

ikethecoder committed:
upd docker compose for two different keycloak config

ikethecoder committed:
upd docker compose for two different keycloak config

ikethecoder committed:
upd docker compose for two different keycloak config

ikethecoder committed:
get logging to appear in tests

ikethecoder committed:
upd ghaction

ikethecoder committed:
install chrome browser for playwright

ikethecoder committed:
align the ports so playright inside and outside docker run ok

ikethecoder committed:
upload results in github action

ikethecoder committed:
fix uploaded artifacts

ikethecoder committed:
adj report location

ikethecoder committed:
adj report location for test results

ikethecoder committed:
adj report location for test results

ikethecoder committed:
upd action

ikethecoder committed:
upd readme for plugins

ikethecoder committed:
include image for oidc docs

ikethecoder committed:
try and get test report generated

ikethecoder committed:
upd to ghaction

ikethecoder committed:
upd to ghaction

ikethecoder committed:
resolve upload artifact issue

ikethecoder committed:
fix pipeline for detecting failure

ikethecoder committed:
use the versions in ghaction

ikethecoder committed:
use the versions in ghaction

ikethecoder committed:
fix failing tests due to async validator issue

ikethecoder committed:
change matrix for test scope

ikethecoder committed:
upd docs

ikethecoder committed:
adj matrix

ikethecoder committed:
pass all the linting

ikethecoder committed:
limit matrix for build

ikethecoder committed:
fix typo in jwt-keycloak plugin

ikethecoder committed:
readme updates

ikethecoder committed:
upd readme make allowed iss generic

rustyjux commented on testsuite/tests/plugins/oidc/README.md:
Not all of these are still TODO, right?

ikethecoder replied:
I took out the TODO heading - I haven't identified what has or hasn't been covered - just listed scope.

rustyjux reviewed (approved):
Thanks for the work setting this up. The `runE2Etest` function took me a little to wrap my head around but seems like it should work well for the narrow context of plugin testing.

A couple general comments:
- docs: seems like the docs in `/docs` come from the source repos. It's a bit confusing having these in a separate docs folder - do I look at those docs or the READMEs? Maybe the files in `/docs` could be moved into the appropriate folders in `/plugins`. Or if they aren't that relevant/useful (and one should instead defer to the included READMEs), maybe just having a link at the top of the README is sufficient
- extensive logging in playwright tests is presumably helpful for debugging if something fails, but it makes the test output long and kinda chaotic. I can understand why it would be simplest to leave always on for the limited length and frequency of these test runs, but a logger utility would be nice to have.

ikethecoder committed:
remove old docs add pino logger

ikethecoder committed:
upd readme

ikethecoder commented:
Added `pino` logger and removed old docs - so PINO_LOG_LEVEL=debug will dump a bunch of stuff out but otherwise it will just be the playwright summary progress.

ikethecoder committed:
minor cleanup of test ghaction

ikethecoder committed:
wait a bit longer for this particular check as it seems to fail in ghaction periodically

ikethecoder committed:
adj the ghaction to run test on main and feature branch
